content/news/2022-01-06-bevy-0.6/index.md

```diff
@@ -0,0 +1,1007 @@
+* **Automatic registration**: Moving more logic into Component also gives us the ability to do fancier things in the future like "automatically registering Reflect impls when deriving Component". Non-blanket Component impls do add a small amount of boilerplate, but they also have the potential to massively reduce the "total boilerplate" of an app.
+* **Documentation**: Deriving Component serves as a form of self-documentation. Its now easy to tell what types are components at a glance.
+* **Decentralized**: In Bevy 0.5 Component-specific configuration like "storage type" had to be registered in a centralized Plugin somewhere. Moving Component configuration into the Component trait decentralizes things and allows users to keep "Component type information" right next to the type itself. 
+* **Event Handlers**: Non-blanket Component impls allow us to add event handlers like `on_insert(world: &mut World)` to the Component trait. Very useful! 
+
+Hopefully by now you're convinced that this is the right move. If not ... I'm sorry ... you still need to implement Component manually in Bevy 0.6. You can either derive Component:
+
+```rust
+// defaults to "Table" storage
+#[derive(Component)]
+struct SomeComponent;
+
+// overrides the default storage
+#[derive(Component)]
+#[component(storage = "SparseSet")]
+struct SomeComponent;
+```
+
+Or you can manually implement it:
+
+```rust
+struct SomeComponent;
+
+impl Component for SomeComponent {
+    type Storage = TableStorage;
+}
+
+impl Component for Movable {
+    type Storage = SparseSetStorage;
+}
+```
+
+### iter() for mutable Queries
+
+<div class="release-feature-authors">authors: @Guvante</div>
+
+Mutable queries can now be immutably iterated (which returns immutable references):
+
+```rust
+fn system(mut players: Query<&mut Player>) {
+    for player in players.iter() {
+        // player is an immutable reference
+    }
+
+    for mut player in players.iter_mut() {
+        // player is a mutable reference
+    }
+}
+```
+
+Compare that to the complicated QuerySet that this would have needed in previous Bevy versions to avoid conflicting immutable and mutable Queries:
+
+```rust
+// Gross!
+fn system(mut players: QuerySet<(QueryState<&Player>, QueryState<&mut Player>)>) {
+    for player in players.q0().iter() {
+        // player is an immutable reference
+    }
+
+    for mut player in players.q1().iter_mut() {
+        // player is a mutable reference
+    }
+}
+```
+
+### SystemState
+
+Have you ever wanted to use "system params" directly with a Bevy World? With {{rust_type(type="struct" crate="bevy_ecs" version="0.6.0" name="SystemState")}}, now you can!
```

Comment:
This is underhyped. SystemState is an absolute game-changer for how easy it is to work with the `World` for any non-trivial piece of code, because it handles the split mutability for you using a familiar API.

Reply:
Added more hype!